### PR TITLE
1107 mass edit line header dates

### LINKED ISSUE #1107: LineView: mass edit lineHeader start dates frontend part
FE PR: https://github.com/HSLdevcom/jore-map-ui/pull/1120
BE PR: https://gitlab.hsl.fi/jore/jore-map-backend/merge_requests/139

At page: https://dev.jore-map.hsldev.com/line/1074

enable mass editing of dates

also:

* make lineHeader as a required date
* hide "lineHeader" section when creating a new line

### PR DESCRIPTION
**Pull request by Divinare**
Closes #1107

BE PR: https://gitlab.hsl.fi/jore/jore-map-backend/merge_requests/139

Frontend part of mass editing lineViews

Also fixed calendar issue where calendar near sidebar was drawn under map. Had to switch calendar libary to get that fixed.

Saving continues in #1122 
MapLayersControl hover menu caldendar close bug fix: #1123

### CODE REVIEW

File: src/components/controls/InputContainer.tsx [hunk 1]
```diff
@@ -67,6 +73,99 @@ const renderEditableContent = (props: IInputProps) => {
...
+}: {
+    onChange: (value: any) => void;
+    placeholder: string;
+    value?: Date;
+    isClearButtonVisible?: boolean;
+}) => {
+    const onInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+        // TODO: implement a better way of changing input value
+        const value = event.target.value;
+        if (Moment(value, 'DD.MM.YYYY', true).isValid()) {
```
Comment (ahjyrkia) [inline src/components/controls/InputContainer.tsx]: It don't seem to make any difference if lines 142-145 weren't there but then again it has TODO comment so I don't know if changes are necessary

File: src/components/sidebar/lineView/lineHeaderTable.scss [hunk 2]
```diff
@@ -1,44 +1,82 @@
...
-        padding: 3px;
+        .lineHeaderButton {
+            display: flex;
+            font-size: $largeFontSize;
+            padding: 5px;
+            margin: 4px;
+            margin-bottom: auto;
+        }
+
+        .removeLineHeaderButton {
```
Comment (ahjyrkia) [inline src/components/sidebar/lineView/lineHeaderTable.scss]: I would have the remove button blue but keep the background color as it is now (white icon and red background when hovering)

File: src/components/sidebar/lineView/lineHeaderTable.scss [hunk 3]
```diff
@@ -1,44 +1,82 @@
...
 
         .lineHeaderRowHighlight {
             background-color: $lightblue;
         }
-    }
 
-    .editLineHeaderButton {
-        font-size: 1.8em;
-        padding: 3px;
+        .lineHeaderButton {
```
Comment (ahjyrkia) [inline src/components/sidebar/lineView/lineHeaderTable.scss]: border-radius: 5px would look nice here

**Comment by Divinare** [comment]
Review fixes done + added a card https://github.com/HSLdevcom/jore-map-ui/issues/1140

**Comment by ahjyrkia** [comment]
Review done.

**Pull request merged**